### PR TITLE
First version

### PR DESCRIPTION
**Pull request by iilyak**

### CODE REVIEW

**Comment by rnewson** [comment]
can you squash? not necessarily to one but to whatever makes sense.

File: test/couch_epi_data_source_tests.erl [hunk 1]
```diff
@@ -0,0 +1,78 @@
+-module(couch_epi_data_source_tests).
```
Comment (kxepal) [inline test/couch_epi_data_source_tests.erl:13]: License header is missed.

File: test/couch_epi_functions_tests.erl [hunk 2]
```diff
@@ -0,0 +1,114 @@
+-module(couch_epi_functions_tests).
```
Comment (kxepal) [inline test/couch_epi_functions_tests.erl:13]: License header is missed.

File: test/couch_epi_tests.erl [hunk 3]
```diff
@@ -0,0 +1,491 @@
+-module(couch_epi_tests).
```
Comment (kxepal) [inline test/couch_epi_tests.erl:13]: License header is missed.

File: test/couch_epi_tests.erl [hunk 4]
```diff
@@ -0,0 +1,491 @@
...
+        ?assertMatch(error, get(Ctx, is_called))
+    end).
+
+
+pipe_state(Ctx) ->
+    Trace = [get(Ctx, inc1), get(Ctx, inc2)],
+    lists:usort([State || {ok, State} <- Trace]).
+
+check_dump(_Module, #ctx{handle = Handle}) ->
+    ?_test(begin
```
Comment (kxepal) [inline test/couch_epi_tests.erl]: `?_test(?assertX(...))` is [actually](https://github.com/erlang/otp/blob/maint/lib/eunit/include/eunit.hrl#L123-L138) `?_assertX(...)`

**Comment by kxepal** [comment]
Great work!

**Comment by iilyak** [comment]
@kxepal: Thank you. Licensing info is added. 
@rnewson: PR is squashed.

**Comment by rnewson** [comment]
@iilyak merged to asf, congrats and great work!

**Pull request merged**